algorithms/kernel/adaboost/adaboost_train_impl.i

```diff
@@ -315,10 +314,10 @@ services::Status AdaBoostTrainKernelNew<method, algorithmFPType, cpu>::adaboostS
     DAAL_CHECK(aErrFlag.get(), services::ErrorMemoryAllocationFailed);
 
     /* Initialize weights */
-    for (size_t i = 0; i < nVectors; i++)
+    threader_for(nVectors, nVectors, [&](const size_t i)
```

Comment:
I think this type of threader_for leads to huge overhead.

Reply:
It depends on the number of vectors nad it was the request from Mikhail.

Reply:
My concern is that you create nVectors tasks.
In this case you should use this function already optimized:
T* service_memset(T * const ptr, const T value, const size_t num)